selfdrive/car/nissan/interface.py

```diff
@@ -77,9 +76,9 @@ def update(self, c, can_strings):
     events = self.create_common_events(ret)
 
     if self.CS.lkas_enabled:
-      events.append(create_event('invalidLkasSetting', [ET.PERMANENT]))
+      events.add(car.CarEvent.EventName.invalidLkasSetting)
 
-    ret.events = events
+    events.to_msg(ret)
```

Comment:
I don't like this pattern everywhere, you really shouldn't modify what you pass in.

ret.events = events.to_msg() is a lot more readable

Reply:
I thought it'd be faster to avoid the python list and keep everything as a capnp list, especially since this is a high frequency call. I profiled it though, and turns out it doesn't make much of a difference